com.unity.netcode.gameobjects/Tests/Editor/Messaging/MessageRegistrationTests.cs

```diff
@@ -179,5 +179,92 @@ public void WhenCreatingMessageSystem_BoundTypeMessageHandlersAreRegistered()
                 Assert.AreEqual(handlerFour, systemThree.MessageHandlers[systemThree.GetMessageType(typeof(TestMessageFour))]);
             }
         }
+
+        internal class AAAEarlyLexicographicNetworkMessage : INetworkMessage
+        {
+            public void Serialize(FastBufferWriter writer)
+            {
+            }
+
+            public bool Deserialize(FastBufferReader reader, ref NetworkContext context)
+            {
+                return true;
+            }
+
+            public void Handle(ref NetworkContext context)
+            {
+            }
+        }
+
+#pragma warning disable IDE1006
+        internal class zzzLateLexicographicNetworkMessage : AAAEarlyLexicographicNetworkMessage
+        {
+        }
+#pragma warning restore IDE1006
+
+        internal class OrderingMessageProvider : IMessageProvider
+        {
+            public List<MessagingSystem.MessageWithHandler> GetMessages()
+            {
+                var listMessages = new List<MessagingSystem.MessageWithHandler>();
+
+                var messageWithHandler = new MessagingSystem.MessageWithHandler();
+
+                messageWithHandler.MessageType = typeof(zzzLateLexicographicNetworkMessage);
+                listMessages.Add(messageWithHandler);
+
+                messageWithHandler.MessageType = typeof(ConnectionRequestMessage);
+                listMessages.Add(messageWithHandler);
+
+                messageWithHandler.MessageType = typeof(ConnectionApprovedMessage);
+                listMessages.Add(messageWithHandler);
+
+                messageWithHandler.MessageType = typeof(OrderingMessage);
+                listMessages.Add(messageWithHandler);
+
+                messageWithHandler.MessageType = typeof(AAAEarlyLexicographicNetworkMessage);
+                listMessages.Add(messageWithHandler);
+
+                return listMessages;
+            }
+        }
+
+        [Test]
```

Comment:
Would be nice to see some test cases for negative cases (e.g. passing out of range params, reordering to the same spot it's already in, etc.)